native_client/generate_trie.cpp

```diff
@@ -0,0 +1,66 @@
+#include <algorithm>
+#include <iostream>
+#include <string>
+using namespace std;
+
+#include "lm/model.hh"
+#include "trie_node.h"
+#include "alphabet.h"
+
+typedef lm::ngram::ProbingModel Model;
+
+lm::WordIndex GetWordIndex(const Model& model, const std::string& word) {
+  lm::WordIndex vocab;
+  vocab = model.GetVocabulary().Index(word);
+  return vocab;
+}
+
+float ScoreWord(const Model& model, lm::WordIndex vocab) {
+  Model::State in_state = model.NullContextState();
+  Model::State out;
+  lm::FullScoreReturn full_score_return;
+  full_score_return = model.FullScore(in_state, vocab, out);
```

Comment:
Why is there this `out` that we don't use ?

Reply:
Asking because according to that comment https://github.com/mozilla/DeepSpeech/pull/805/commits/f1e859ef7b94ec01dcea67d91b7dfe7d271fe80c#diff-7a0b3b152c06fc03b20e63b72c26583aR11 (which I don't completely get the point, so I might just be misunderstanding something there), it seems like `out_state` has some use.

Reply:
Keeping state is a performance optimization when scoring a sentence of various words/n-grams, in this function we want to score the word independently of any context so that's why we throw it away.

Reply:
Thanks, maybe worth a comment then ?